pennylane_lightning/core/src/simulators/lightning_qubit/gates/cpu_kernels/GateImplementationsLM.hpp

```diff
@@ -1832,6 +1836,51 @@ class GateImplementationsLM : public PauliGenerator<GateImplementationsLM> {
         }
     }
 
+    template <class PrecisionT, class ParamT>
+    static void
+    applyGlobalPhase(std::complex<PrecisionT> *arr, std::size_t num_qubits,
+                     [[maybe_unused]] const std::vector<std::size_t> &wires,
+                     [[maybe_unused]] bool inverse,
+                     [[maybe_unused]] ParamT angle) {
+        applyNCGlobalPhase(arr, num_qubits, {}, {}, wires, inverse, angle);
+    }
+
+    template <class PrecisionT, class ParamT>
+    static void
+    applyNCGlobalPhase(std::complex<PrecisionT> *arr, std::size_t num_qubits,
+                       const std::vector<std::size_t> &controlled_wires,
+                       const std::vector<bool> &controlled_values,
+                       [[maybe_unused]] const std::vector<std::size_t> &wires,
+                       bool inverse, ParamT angle) {
+        const std::complex<PrecisionT> phase =
+            std::exp(std::complex<PrecisionT>(0, inverse ? angle : -angle));
+        auto core_function = [&phase](std::complex<PrecisionT> *arr,
+                                      const std::size_t i0,
+                                      const std::size_t i1) {
+            arr[i0] *= phase;
+            arr[i1] *= phase;
+        };
+        std::size_t target{0};
+        if (!controlled_wires.empty()) {
+            for (std::size_t i = 0; i < num_qubits; i++) {
+                if (std::find(controlled_wires.begin(), controlled_wires.end(),
+                              i) == controlled_wires.end()) {
```

Comment:
Is calling find in a loop like this ok for the usage we expect?

Reply:
I didn't like that piece of code either, but it should not impact performance in any case. The problem is the following: 

- `controlled_wires` is a small array with length at most `num_qubits - 1`.
- we want a wire, any wire, that's not in `controlled_wires` and set the wire for the kernel to that.

Can you think of a nice 1-/2-liner to get this on top of your head?

Reply:
Based on that, I think for now I think this is fine. If the order was sorted, it'd be faster to do bin-search, but while this isn't a bottleneck let's go for it.